aspnetcore/fundamentals/configuration.md

```diff
@@ -1,166 +1,357 @@
 ---
-title: Configuration
+title: Configuration | Microsoft Docs
 author: rick-anderson
+description: Demonstrates the configuration API
+keywords: ASP.NET Core, configuration, JSON
 ms.author: riande
 manager: wpickett
-ms.date: 10/14/2016
+ms.date: 11/29/2016
 ms.topic: article
 ms.assetid: b3a5984d-e172-42eb-8a48-547e4acb6806
+ms.technology: aspnet
 ms.prod: aspnet-core
 uid: fundamentals/configuration
 ---
-# Configuration
-
 <a name=fundamentals-configuration></a>
 
-[Steve Smith](http://ardalis.com), [Daniel Roth](https://github.com/danroth27)
+  # Configuration
+
+[Rick Anderson](https://twitter.com/RickAndMSFT), [Mark Michaelis](http://intellitect.com/author/mark-michaelis/), [Steve Smith](http://ardalis.com), [Daniel Roth](https://github.com/danroth27)
+
+<!-- SEE https://github.com/aspnet/Configuration/issues/532
+-->
+
+The configuration API reads lists of name-value pairs. The name-value pairs can be grouped into a multi-level hierarchy. There are configuration providers for:
+
+* File formats (INI, JSON, and XML)
+* Command-line arguments
+* Environment variables
+* In-memory .NET objects
+* An encrypted user store
+* Custom providers, which you install or create
 
-ASP.NET Core supports a variety of different configuration options. Application configuration data can come from files using built-in support for JSON, XML, and INI formats, as well as from environment variables, command line arguments or an in-memory collection. You can also write your own [custom configuration provider](xref:fundamentals/configuration#custom-config-providers).
+Each configuration value maps to a string key. There’s built-in binding support to deserialize settings into a custom [POCO](https://en.wikipedia.org/wiki/Plain_Old_CLR_Object) object (.NET class).
 
 [View or download sample code](https://github.com/aspnet/docs/tree/master/aspnetcore/fundamentals/configuration/sample)
 
-## Getting and setting configuration settings
+  ## Simple configuration
 
-ASP.NET Core's configuration system has been re-architected from previous versions of ASP.NET, which relied on `System.Configuration` and XML configuration files like `web.config`. The new configuration model provides streamlined access to key/value based settings that can be retrieved from a variety of sources. Applications and frameworks can then access configured settings in a strongly typed fashion using the new [Options pattern](xref:fundamentals/configuration#options-config-objects).
+The following console app uses the JSON configuration provider:
 
-To work with settings in your ASP.NET application, it is recommended that you only instantiate a `Configuration` in your application's `Startup` class. Then, use the [Options pattern](xref:fundamentals/configuration#options-config-objects) to access individual settings.
+[!code-csharp[Main](configuration/sample/src/ConfigJson/Program.cs)]
```

Comment:
This is regarding the code snippet: It looks a bit complicated given that the title says "Simple configuration" :smile: The main issue is actually the commented code. Is there any way to simplify it further? BTW, I think you should refer to .NET 4.x as .NET Framework 4.x.